packages/appsync-modelgen-plugin/src/visitors/appsync-dart-visitor.ts

```diff
@@ -237,6 +285,37 @@ export class AppSyncModelDartVisitor<
     return classDeclarationBlock.string;
   }
 
+  protected generateNonModelClass(model: CodeGenModel): string {
+    const includeIdGetter = false;
+    //class wrapper
+    const classDeclarationBlock = new DartDeclarationBlock()
+      .asKind('class')
+      .withName(this.getModelName(model))
+      .withComment(`This is an auto generated class representing the ${model.name} type in your schema.`)
+      .annotate(['immutable']);
+    //model fields
+    model.fields.forEach(field => {
+      this.generateModelField(field, '', classDeclarationBlock);
+    });
+    //getters
```

Comment:
nit: these comments don't add to much value, the intent is already in the method names

Reply:
Thanks and agreed. I followed existing code pattern when implemented this method... I can sync up with codegen maintainers and help cleaning up separately.